### DISCUSSION TITLE
SolutionUserObject and getMaterialPropertyOldByName

### DISCUSSION THREAD

**Discussion by TheGreatCid** [idaholab/moose#26065] [Q&A General] [answered]
Hi all,

I am using a SolutionUserObject to populate variables/materials and start a simulation. I have a material that requires both the current and old value.  
So it is declared as:
```
_ep(declareADProperty<Real>(prependBaseName("effective_plastic_strain"))),
_ep_old(getMaterialPropertyOldByName<Real>(prependBaseName("effective_plastic_strain")))
```
I have the timestep I want to load the primary data from and the previous one. 
Is it possible to populate both fields and restart the simulation as usual? 

I remember looking into this long ago and finding that MOOSE didn't have this capability, but I wanted to see if anything new has been developed for this application. 

Also, in general, if a material property is declared in the source code rather than in the input file, is it possible to populate the field from a SolutionUserObject, or will I need to rewrite the code to `getMaterialPropertyByName` and define the material in the input file?

Thanks!

**Comment by lindsayad** [comment]
Can you use checkpoint restart instead?

**Reply by TheGreatCid** [reply]
I cannot because I am applying a re-meshing algorithm outside of moose. So the new mesh will not match what is in the checkpoint

**Reply by lindsayad** [reply, marked as answer]
If you've investigated this in the past, then I suspect you know `SolutionUserObject` capabilities better than I do. I can tell you that there have been no recent developments in that class related to materials

**Reply by TheGreatCid** [reply]
Got it, thanks

**Comment by GiudGiud** [comment]
Hello

So the SolutionUO can be used for ICs, functions, variables etc but we have not made a SolutionMaterial or an interface that lets you get the values of a stateful material from a solution file (even though they are definitely in there, as we need them when we do checkpoint restart).

You would definitely need to change the code in the material. The 'getMaterialPropertyOld' will not know to query the solutionUO for those values. 

If that is acceptable to you then there's two tasks to do :
- make a SolutionFileMaterial
- create an API to load an 'old' value for a material property from another material property

Guillaume

**Reply by GiudGiud** [reply]
I can create a feature request for you, please let me know
However we do not have any available resources at the time so the delay would be significant if we implemented it

**Reply by TheGreatCid** [reply]
This would be quite a useful feature to have, so a feature request would be much appreciated! 
In the meantime, I have an idea of how to implement this feature in an inelegant way to suit what I need to do.
